src/lib.rs

```diff
@@ -28,11 +28,13 @@ mod util;
 
 #[doc(include = "../docs/notes.md")]
 mod notes {}
+mod circuit_proof;
 mod generators;
 mod inner_product_proof;
 mod proof_transcript;
 mod range_proof;
 
+pub use circuit_proof::CircuitProof;
```

Comment:
Maybe we should leave the `circuit_proof` code unexported until we figure out what kind of API we want people to use for circuit proving?

My intuition so far is that we probably don't want to expose an API like the one we have now, so if we keep the modules unexported we can work towards having a releasable library for range proofs, without having to commit to a circuit proving API before it's done.